### PR TITLE
Add support for aws filter plugin

### PR DESCRIPTION
**Pull request by wenchajun**
### What this PR dose?
Adds support for https://docs.fluentbit.io/manual/pipeline/filters/aws-metadata filter plugin

### Why we need it?
Fluentbit supports a variety of plug-ins, but not all fluentbit-operator adaptations. More plug-ins will be supported in the future, see the fields supported by the Filter plug-in
| Field | Description | Scheme |Default|
| ----- | ----------- | ------ | -----|
| imds_version | Specify which version of the instance metadata service to use. Valid values are 'v1' or 'v2'. | string |v2|
| az | The availability zone; for example, "us-east-1a". | bool |true|
| ec2_instance_id | The EC2 instance ID. | bool |true|
| ec2_instance_type | The EC2 instance type. | bool |false|
| private_ip | The EC2 instance private ip.| bool |false|
| ami_id | The EC2 instance image id. | bool |false|
| account_id | The account ID for current EC2 instance. | bool |false|
| hostname | The hostname for current EC2 instance.| bool |false|
| vpc_id | The VPC ID for current EC2 instance. | bool |false|

### The final results
The configuration is written to secret

Signed-off-by: chengdehao <dehaocheng@yunify.com>

### CODE REVIEW

File: api/fluentbitoperator/v1alpha2/plugins/filter/aws_types.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+// The AWS Filter Enriches logs with AWS Metadata. Currently the plugin adds the EC2 instance ID and availability zone to log records.
+type Aws struct {
+	// Specify which version of the instance metadata service to use. Valid values are 'v1' or 'v2'.
+	// +kubebuilder:validation:Enum:=v1;v2
+	ImdsVersion string `json:"imdsVersion,omitempty"`
+	// The availability zone; for example, "us-east-1a". Default is true.
+	Az *bool `json:"az,omitempty"`
+    //The EC2 instance ID.Default is true.
+	Ec2InstanceId  *bool `json:"ec2InstanceId,omitempty"`
```
Comment (wanjunlei) [inline api/fluentbitoperator/v1alpha2/plugins/filter/aws_types.go:19]: Ec2InstanceId => Ec2InstanceID
PrivateIp => PrivateIP
AmiId => AmiID
AccountId => AccountID
VpcId => VpcID

**Review by wanjunlei — CHANGES_REQUESTED** [review]

File: api/fluentbitoperator/v1alpha2/plugins/filter/aws_types.go (hunk 1 above)
Comment (wenchajun) [reply api/fluentbitoperator/v1alpha2/plugins/filter/aws_types.go]: Thanks, fixed

File: api/fluentbitoperator/v1alpha2/plugins/filter/aws_types.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	"fmt"
+
+	"kubesphere.io/fluentbit-operator/api/fluentbitoperator/v1alpha2/plugins"
+	"kubesphere.io/fluentbit-operator/api/fluentbitoperator/v1alpha2/plugins/params"
+)
+
+// +kubebuilder:object:generate:=true
+
+// The AWS Filter Enriches logs with AWS Metadata. Currently the plugin adds the EC2 instance ID and availability zone to log records.
+type Aws struct {
```
Comment (wanjunlei) [inline api/fluentbitoperator/v1alpha2/plugins/filter/aws_types.go:13]: Aws => AWS

File: api/fluentbitoperator/v1alpha2/filter_types.go [hunk 3]
```diff
@@ -61,6 +61,8 @@ type FilterItem struct {
 	Throttle *filter.Throttle `json:"throttle,omitempty"`
 	// RewriteTag defines a RewriteTag configuration.
 	RewriteTag *filter.RewriteTag `json:"rewriteTag,omitempty"`
+	//Aws defines a Aws configuration.
+	Aws *filter.Aws `json:"aws,omitempty"`
```
Comment (wanjunlei) [inline api/fluentbitoperator/v1alpha2/filter_types.go:65]: Aws => AWS

**Review by wanjunlei — CHANGES_REQUESTED** [review]

File: api/fluentbitoperator/v1alpha2/plugins/filter/aws_types.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+// The AWS Filter Enriches logs with AWS Metadata. Currently the plugin adds the EC2 instance ID and availability zone to log records.
+type AWS struct {
+	// Specify which version of the instance metadata service to use. Valid values are 'v1' or 'v2'.
+	// +kubebuilder:validation:Enum:=v1;v2
+	ImdsVersion string `json:"imdsVersion,omitempty"`
+	// The availability zone; for example, "us-east-1a". Default is true.
+	Az *bool `json:"az,omitempty"`
+	//The EC2 instance ID.Default is true.
+	EC2InstanceID *bool `json:"ec2InstanceId,omitempty"`
```
Comment (wanjunlei) [inline api/fluentbitoperator/v1alpha2/plugins/filter/aws_types.go:20]: ec2InstanceId => ec2InstanceID
privateIp => privateIP
amiId => amiID
accountId => accountID
vpcId => vpcID

Comment (wenchajun) [reply api/fluentbitoperator/v1alpha2/plugins/filter/aws_types.go]: Sorry, I overlooked that

File: api/fluentbitoperator/v1alpha2/plugins/filter/aws_types.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+// +kubebuilder:object:generate:=true
+
+// The AWS Filter Enriches logs with AWS Metadata. Currently the plugin adds the EC2 instance ID and availability zone to log records.
+type AWS struct {
+	// Specify which version of the instance metadata service to use. Valid values are 'v1' or 'v2'.
+	// +kubebuilder:validation:Enum:=v1;v2
+	ImdsVersion string `json:"imdsVersion,omitempty"`
+	// The availability zone; for example, "us-east-1a". Default is true.
+	Az *bool `json:"az,omitempty"`
```
Comment (benjaminhuo) [inline api/fluentbitoperator/v1alpha2/plugins/filter/aws_types.go:18]: Az could be changed to AZ too

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**